pkg/da/data_availability_header.go

```diff
@@ -62,9 +65,39 @@ func NewDataAvailabilityHeader(eds *rsmt2d.ExtendedDataSquare) (DataAvailability
 	return dah, nil
 }
 
+// ConstructEDS constructs an ExtendedDataSquare from the given transactions and app version.
+// If maxSquareSize is less than 0, it will use the upper bound square size for the given app version.
+func ConstructEDS(txs [][]byte, appVersion uint64, maxSquareSize int) (*rsmt2d.ExtendedDataSquare, error) {
```

Comment:
Okay so for celestia-node's call to this function, we should always pass -1 for `maxSquareSize` such that this function sets the defaults for us. 👍🏻